### PR TITLE
Port to Oreo

### PR DESCRIPTION
**Pull request by AdamNiederer**
Register a notification channel for the overlay notification, add a build
flavor, and use newer versions of Gradle and build tools

Tested with a Nougat device and Oreo emulator

### CODE REVIEW

**Comment by smichel17** [comment]
Thanks, I appreciate this! I have a really busy week right now so I probably won't find time to review this until next Monday or Tuesday.

**Comment by AdamNiederer** [comment]
No problem. I'll try to fiddle with Travis to make it build in the mean time.

**Comment by smichel17** [comment]
Hi, sitting down to look at this now. While you're here, I'd appreciate if you wanted to share anything you found challenging about getting into the Red Moon code base. Long-term, I'm interested in getting another maintainer or few, to make sure the project doesn't die if something happens to my availability. That starts with people making small contributions, so I want to make that as easy as possible. Thanks!

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -187,4 +187,8 @@
     <string name="time_toggle_options_array_0">Manual</string>
     <string name="time_toggle_options_array_1">Automatic (sun)</string>
     <string name="time_toggle_options_array_2">Automatic (custom)</string>
+
+    <!-- Notification channel descriptions -->
+    <string name="notification_channel_overlay_name">Overlay Control</string>
+    <string name="notification_channel_overlay_description">The persistent overlay control notification</string>
```
[the remark this answers is not in the document]
Comment (AdamNiederer) [reply app/src/main/res/values/strings.xml]: These show up in the new notification settings in Oreo. The first one is the human-readable title of the channel, and the second one is a longer description which appears the the bottom of the page. The path is `App Info > App notifications > Overlay Control`

I'll replace "overlay" with "filter" shortly.

screenshot_1527038768

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -37,11 +37,13 @@
...
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.jmstudios.redmoon" >
+    xmlns:tools="http://schemas.android.com/tools"
+    package="com.jmstudios.redmoon">
 
     <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW" />
-    <uses-permission android:name="android.permission.WRITE_SETTINGS" />
+    <uses-permission android:name="android.permission.WRITE_SETTINGS"
+        tools:ignore="ProtectedPermissions" />
```
[the remark this answers is not in the document]
Comment (AdamNiederer) [reply app/src/main/AndroidManifest.xml]: It's coming up as a lint on mine (Studio 3.1.2, 27r03). I'll try running it through without it.

2018-05-22-213552_680x34_scrot

[the remark this answers is not in the document]
Comment (AdamNiederer) [reply app/src/main/AndroidManifest.xml]: Yep, looks like it won't pass Travis without the lint suppression.

**Comment by AdamNiederer** [comment]
I've addressed each comment, and threw in a few boring IDE fixes. I just flashed my phone to Oreo, so I only tested 1537b7b on an Oreo device and Oreo emulator.

**Comment by smichel17** [comment]
Sorry this took so long to merge. I was going to clean up a few things, but then forgot. Figure I'll merge now and clean up after :)

I also added you as a collaborator for the repo, so you can make tweaks if you find something that's broken. Feel free to push directly or open PRs if you want me to look it over, first.

**Comment by smichel17** [comment]
I'll also add: I'm willing to walk you through the whole code base, if you're interested. It would probably get me back into working on Red Moon more, too :P

**Pull request merged**